src/components/FooterBar/FooterBar.tsx

```diff
@@ -90,21 +104,35 @@ const bottomRender = ({ label, to }: FooterColumnItemProps): React.ReactElement
   </a>
 );
 
-export const FooterBar: FunctionComponent = () => {
-  const { configFile } = usePersistedConfigFile();
-  const data = getData(configFile);
-  const networkPath = getNetworkPath(configFile?.network);
-  const legalData = {
+const legalData = (networkPath: string) => {
+  return {
     copyright: "Copyright \u00A9 2021 TradeTrust",
     items: [
       { label: "Privacy Policy", to: `${networkPath}/privacy-policy`, render: bottomRender },
       { label: "Terms of use", to: `${networkPath}/terms-of-use`, render: bottomRender },
     ],
   };
+};
+
+export const FooterBar: FunctionComponent = () => {
+  const { configFile } = usePersistedConfigFile();
+  const [data, setData] = useState<footerData[]>();
+  const [networkPath, setNetworkPath] = useState<string>("https://tradetrust.io");
+
+  useCallback(() => {
+    const networkP = getNetworkPath(configFile?.network);
+    setData(getData(networkP));
+    setNetworkPath(networkP);
+  }, [configFile]);
```

Comment:
umm... Any reasons for doing this?

Reply:
so that the component will only re render when the config file change.

Reply:
i forgot to change it back to useEffect.

Reply:
i reverted back to without this.

Reply:
Okiee.👌